### ISSUE TITLE
How does --lint-fsatrace work?

### ISSUE THREAD

**Issue by peti** [ndmitchell/shake#573]
I have a shake build system that I'd like to run with `fsatrace`-style linting. However, I cannot diagnose whether that attempt actually works or not! When I run the command

~~~
$ cabal -v0 new-run update-emu-data -- --verbose --lint-fsatrace --rebuild
# writeFile (for _build/active-emus.json)
# writeFile (for _build/emergency.html)
Lint checking succeeded
Build completed in 0:05m
~~~

then everything seems to work fine, but in fact I ran that command without `fsatrace` in `$PATH`:

~~~
$ type -pa fsatrace
fsatrace not found
~~~

So how come that build was successful? Shouldn't that build have failed?

**Comment by ndmitchell** [comment]
Are you calling any `cmd` things? the `fsatrace` linting only captures when you run external processes, and from the trace above, it looks like you are only doing `writeFile`. I should document that.

**Comment by peti** [comment]
Yeah, apparently I misunderstood. I assumed that somehow any and all file accesses performed by the shake build would be tracked, but in hindsight it make sense that only external command calls are wrapped by `fsatrace`. Thanks!

**Comment by ndmitchell** [comment]
Needs documenting for sure!
